home/private_dot_local/lib/agent-equipment/agent_equipment/validator.py

```
@@ -140,6 +154,116 @@ def canonical_json_sha256(document: JsonObject) -> str:
+
+
+def _hold_bounded_document(
+    descriptors: ExitStack,
+    path: Path,
+    *,
+    maximum_bytes: int,
+) -> _HeldDocument:
+    no_follow = getattr(os, "O_NOFOLLOW", None)
+    nonblocking = getattr(os, "O_NONBLOCK", None)
+    if type(no_follow) is not int or type(nonblocking) is not int:
+        raise OSError("safe nonblocking document reads are unavailable")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)